security/pkg/nodeagent/cache/monitoring.go

```diff
@@ -55,3 +60,13 @@ func init() {
 		numFileSecretFailures,
 	)
 }
+
+func RecordCertExpiry(resourceName string, expiry time.Time) {
+	monitoring.NewDerivedGauge(
```

Comment:
Can we add labels to derived metric? I know it is currently supported but I don't think there should be anything blocking?

Reply:
Feels wrong to have resource name at stats name.

Reply:
need to take a look at istio/pkg package

Reply:
looks good, need to update `istio/pkg` first.

```console
# HELP istio_agent_cert_expiry_seconds The left time, in seconds, when cert chain will expire. A negative value indicates the cert is expired.
# TYPE istio_agent_cert_expiry_seconds gauge
istio_agent_cert_expiry_seconds{resource_name="default"} 86108.240811166
```